### PR TITLE
Some improvements ;)

### PR DESCRIPTION
**Pull request by kessiler**

### CODE REVIEW

**Comment by herophuong** [comment]
This pull request has two serious security problems:

1. default to bind to 0.0.0.0
2. fool users to believe it is binding to 127.0.0.1 whenever it is binding to 0.0.0.0

I don't recommend to merge this pull request

**Comment by kessiler** [comment]
@herophuong you can change the bind address to whatever you want. You don't really need to bind it on 0.0.0.0.
There isn't anywhere saying its binding to 0.0.0.0, its telling that you proxy is available at "localhost/127.0.0.1" which is the loopback adapter. Accessing from 0.0.0.0 won't take you to the the es proxy/kibana.

File: index.js [hunk 1]
```diff
@@ -7,41 +7,56 @@ var express = require('express');
...
 var figlet = require('figlet');
+var basicAuth = require('basic-auth-connect');
+var compress = require('compression');
 
 var yargs = require('yargs')
     .usage('usage: $0 [options] <aws-es-cluster-endpoint>')
     .option('b', {
         alias: 'bind-address',
-        default: '127.0.0.1',
+        default: process.env.BIND_ADDRESS || '0.0.0.0',
```
Comment (herophuong) [inline index.js]: > There isn't anywhere saying its binding to 0.0.0.0

This is telling me that the code has changed to bind to 0.0.0.0 as default

File: index.js [hunk 2]
```diff
@@ -123,5 +148,6 @@ console.log(figlet.textSync('AWS ES Proxy!', {
     verticalLayout: 'default'
 }));
 
-console.log('AWS ES cluster available at http://' + BIND_ADDRESS + ':' + PORT);
-console.log('Kibana available at http://' + BIND_ADDRESS + ':' + PORT + '/_plugin/kibana/');
+var address = BIND_ADDRESS === '0.0.0.0' ? '127.0.0.1' : BIND_ADDRESS;
```
Comment (herophuong) [inline index.js]: This one is hiding users from knowing the proxy is binding to 0.0.0.0

File: index.js (hunk 1 above)
Comment (kessiler) [inline index.js]: You can change it, its configurable. :)
When I said its not telling that is binding to 0.0.0.0 is bc there's anywhere with "console.log" or anything else printing "I'm going to bind it to 0.0.0.0"

**Comment by herophuong** [comment]
I think you may be haven't understood what is binding to 0.0.0.0 means. Correct me if I'm wrong though. 

Binding to 0.0.0.0 doesn't mean requests to 0.0.0.0 will access the aws-es-kibana, it means outsiders who have your public or private IP address can access the aws-es-kibana by using your IP. 

This is a local service and thus should not **BY DEFAULT** exposes your AWS to the outside world.

File: index.js (hunk 2 above)
Comment (kessiler) [inline index.js]: We can just add - "console.log('The proxy is bound at 0.0.0.0')"

**Comment by kessiler** [comment]
You're right dude but thats the reason its configurable. We can change it to 127.0.0.1 if it will make you more comfortable.

**Comment by herophuong** [comment]
That would be great. Otherwise, it was a great work that you did. 

P/S: Sorry if my language was a bit strong. Those was urgent things security-wise in my opinion so...

**Comment by kessiler** [comment]
@herophuong all right :)

**Comment by santthosh** [comment]
Thanks @kessiler, the changes have been merged offline

**Pull request closed without merging**